include/fi_util.h

```diff
@@ -68,17 +68,19 @@ enum fi_match_type {
 	FI_MATCH_PREFIX,
 };
 
+enum util_prov_opts {
+	UTIL_TX_SHARED_CTX	= 1 << 0,
+	UTIL_RX_SHARED_CTX	= 1 << 1,
+};
```

Comment:
These should be flags, rather than a named enum.  Please drop the name and see below...